content/docs/esc/integrations/external-secrets-operator.md

```diff
@@ -0,0 +1,489 @@
+
+- Sensitive data is stored and managed by an external service outside the Kubernetes cluster and this leads to better security and compliance posture.
+- External Secrets Operator supports multiple secret providers, so you can use the same operator to manage secrets from different sources.
+- You can take advantage of the advanced features of the secret provider, such as encryption of data at rest and scenarios like secret rotation.
+
+Since version `0.10.0` External Secrets Operator supports Pulumi ESC as a secret provider.
+
+## Prerequisites
+
+In this tutorial, you will learn how to deploy the External Secrets Operator and use it to manage secrets stored in Pulumi ESC. Before you start, make sure you have the following prerequisites:
+
+- the [Pulumi ESC CLI](/docs/esc-cli/)
+- a Kubernetes cluster
+- [kubectl](https://kubernetes.io/releases/download/#kubectl)
+- [helm](https://helm.sh/docs/intro/install/)
+
+## Deploy External Secrets Operator
+
+### Deploy using Helm
+
+#### Install from Helm Chart Repository
+
+```bash
+helm repo add external-secrets https://charts.external-secrets.io
+helm repo update
+
+helm upgrade --install external-secrets external-secrets/external-secrets \
+    --namespace external-secrets \
+    --create-namespace \
+    --wait
+```
+
+#### Create secret containing Pulumi access token
+
+```bash
+kubectl create secret generic pulumi-access-token -from-literal=PULUMI_ACCESS_TOKEN=${PULUMI_ACCESS_TOKEN} \
+    --namespace external-secrets
+```
+
+#### Create ClusterSecretStore
+
+Now you can create a `ClusterSecretStore` resource that will tell External Secrets Operator to use Pulumi ESC as a secret provider.
+
+If you want to limit the access by namespace, you can create a `SecretStore` resource instead, which is scoped to a single namespace.
+
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: secret-store
+spec:
+  provider:
+    pulumi:
+      organization: ${PULUMI_ORG}
+      project: ${PULUMI_PROJECT}
+      environment: ${PULUMI_ENV}
+      accessToken:
+        secretRef:
+          name: pulumi-access-token
+          key: PULUMI_ACCESS_TOKEN
+          namespace: external-secrets
+  EOF
+```
+
+Please replace `${PULUMI_ORG}`, `${PULUMI_PROJECT}`, `${PULUMI_ENV}` with your Pulumi organization, project, and environment names.
+
+For demo purposes, we assume that we already have an ESC environment `my-org/my-project/my-env` with a secret `my-secret` that we want to manage using External Secrets Operator.
+
+```yaml
+values:
+  hello: world
+```
+
+#### Create ExternalSecret
+
+Now you can create an `ExternalSecret` resource that will tell External Secrets Operator to fetch the secret from Pulumi ESC. There is also an `ClusterExternalSecret` resource, which is a cluster scoped resource that can be used to manage ExternalSecret resources in specific namespaces.
```

Comment:
whenever we use K8 specific language like 'ClusterExternalSecret' can we link them to external documentation?